pkg/warmup/warmup.go

```diff
@@ -15,45 +15,34 @@
 package warmup
 
 import (
-	"fmt"
 	"log"
 	"mittens/pkg/grpc"
 	"mittens/pkg/http"
 	"sync"
 	"time"
 )
 
-type Options struct {
+// Warmup holds any information needed for the workers to send requests.
+type Warmup struct {
+	Target             Target
 	MaxDurationSeconds int
 	Concurrency        int
 }
 
-type Warmup struct {
-	target  Target
-	options Options
-}
-
-func NewWarmup(readinessHttpClient http.Client, readinessGrpcClient grpc.Client, httpClient http.Client, grpcClient grpc.Client, options Options, targetOptions TargetOptions) (Warmup, error) {
-	target, err := NewTarget(readinessHttpClient, readinessGrpcClient, httpClient, grpcClient, targetOptions)
-	if err != nil {
-		return Warmup{}, fmt.Errorf("new target: %v", err)
-	}
-	return Warmup{target: target, options: options}, err
-}
-
-func (w Warmup) HttpWarmupWorker(wg *sync.WaitGroup, requests <-chan http.Request, headers map[string]string, requestDelayMilliseconds int, requestsSentCounter *int) {
+// HTTPWarmupWorker sends HTTP requests to the target using goroutines.
+func (w Warmup) HTTPWarmupWorker(wg *sync.WaitGroup, requests <-chan http.Request, headers map[string]string, requestDelayMilliseconds int, requestsSentCounter *int) {
 	for request := range requests {
 		time.Sleep(time.Duration(requestDelayMilliseconds) * time.Millisecond)
 
-		resp := w.target.httpClient.Request(request.Method, request.Path, headers, request.Body)
+		resp := w.Target.httpClient.SendRequest(request.Method, request.Path, headers, request.Body)
 
 		if resp.Err != nil {
 			log.Printf("🔴 Error in request for %s: %v", request.Path, resp.Err)
 		} else {
 			*requestsSentCounter++
 
 			if resp.StatusCode/100 == 2 {
-				log.Printf("🟢 %s response for %s %d ms: %v", resp.Type, request.Path, resp.Duration/time.Millisecond, resp.StatusCode)
+				log.Printf("%s response for %s %d ms: %v", resp.Type, request.Path, resp.Duration/time.Millisecond, resp.StatusCode)
```

Comment:
No more little green badge here? 😢

Reply:
It wouldn't render nicely across different editors and decided to remove that :p